packages/unified/src/utilsWeb.ts

```diff
@@ -0,0 +1,274 @@
+import {
+  assertUnreachable,
+  DendronError,
+  ERROR_STATUS,
+  NoteProps,
+  ProcFlavor,
+} from "@dendronhq/common-all";
+// @ts-ignore
+import rehypePrism from "@mapbox/rehype-prism";
+// @ts-ignore
+import mermaid from "@dendronhq/remark-mermaid";
+import _ from "lodash";
+import link from "rehype-autolink-headings";
+import math from "remark-math";
+// @ts-ignore
+import variables from "remark-variables";
+// @ts-ignore
+import raw from "rehype-raw";
+import slug from "rehype-slug";
+import rehypeStringify from "rehype-stringify";
+import remark from "remark";
+import abbrPlugin from "remark-abbr";
+import footnotes from "remark-footnotes";
+import frontmatterPlugin from "remark-frontmatter";
+import remarkParse from "remark-parse";
+import remark2rehype from "remark-rehype";
+import { Processor } from "unified";
+import { hierarchies } from "./remark";
+import { backlinks } from "./remark/backlinks";
+import { backlinksHover } from "./remark/backlinksHover";
+import { blockAnchors } from "./remark/blockAnchors";
+import { dendronHoverPreview } from "./remark/dendronPreview";
+import { dendronPub } from "./remark/dendronPub";
+import { extendedImage } from "./remark/extendedImage";
+import { hashtags } from "./remark/hashtag";
+// import { noteRefsV2 } from "./remark/noteRefsV2";
+import { userTags } from "./remark/userTags";
+import { wikiLinks } from "./remark/wikiLinks";
+import { DendronASTDest } from "./types";
+import {
+  MDUtilsV5,
+  ProcDataFullOptsV5,
+  ProcDataFullV5,
+  ProcMode,
+  ProcOptsV5,
+} from "./utilsv5";
+
+/**
+ * Special version of MDUtilsV5 to get preview working in the web extension.
+ * This class should eventually be deleted and converged with utilsV5 once
+ * utilsV5 is compatible with EngineV3.
+ */
+export class MDUtilsV5Web {
+  public static procRehypeWeb(
+    data: Omit<ProcDataFullOptsV5, "dest">,
+    opts?: { flavor?: ProcFlavor }
+  ) {
+    const proc = this._procRehype(
+      { mode: ProcMode.FULL, parseOnly: false, flavor: opts?.flavor },
+      data
+    );
+    return proc.use(rehypeStringify);
+  }
+
+  private static setProcData(proc: Processor, opts: Partial<ProcDataFullV5>) {
```

Comment:
these methods look duplicate from utilsv5. can we use the utilsv5 methods where it makes sense?

Reply:
Removed dupes.  I actually hope to delete this `utilsWeb.ts` class as soon as it's possible.